components/raftstore/src/store/async_io/write.rs

```diff
@@ -800,7 +828,12 @@ where
         let mut callback_time = 0f64;
         if notify {
             for (region_id, (peer_id, ready_number)) in &self.batch.readies {
-                self.notifier.notify(*region_id, *peer_id, *ready_number);
+                let mut need_scheduled = false;
```

Comment:
It's very strange to let writer tell peer to reschedule, it feels like though `after_write_hook` is an arbitrary closure, but writer know it has to be snapshot. Whether need to reschedule should be decided by the peer itself, writer only needs to persist change, call the hook and notify peer it's persisted.

Reply:
fixed